### PR TITLE
RPC server: add `add-module` endpoint

### LINKED ISSUE #3444: Add new modules on the fly to rpc-server
WIP issue:

- Add a new module with new rules via rpc call `add-module`. New module must only import modules already loaded in the backend.
- Change the existing RPC calls to take a `module` parameter, which always defaults to the main module, but can be set to any module they added on the fly.

Needed for:

- Circularities.
- Speeding up proofs with sub-executions.

### PR DESCRIPTION
**Pull request by Sereja313**
Fixes #3444 

### Estimate:

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: test/rpc-server/add-module/add/definition.kore [hunk 1]
```diff
@@ -0,0 +1,63 @@
+[topCellInitializer{}(LblinitGeneratedTopCell{}()), org'Stop'kframework'Stop'attributes'Stop'Source{}("Source(/home/jost/work/RV/code/haskell-backend/test/rpc-server/resources/a-to-f/test.k)")]
```
Comment (ana-pantilie) [inline test/rpc-server/add-module/add/definition.kore:1]: Can you add some readme.md files which describe what these tests check?

**Review by ana-pantilie — APPROVED** [review]
I only have a couple of questions, other than that, I think this looks good! Nice job finding that function which allows you to add a new module to the definition.

File: docs/2022-07-18-JSON-RPC-Server-API.md [hunk 2]
```diff
@@ -394,6 +432,37 @@ This error wraps an error message from the internal implication check routine, i
...
+## -32004 Could not parse KORE pattern
+
+This error wraps the internal error thrown when parsing the received plain text module.
+
+```json
+{
+  "jsonrpc":"2.0",
+  "id":1,
+  "error":  {
+    "data":":11:3: unexpected token TokenIdent \\"hooked-IST\\"\\n",
```
[the remark this answers is not in the document]
Comment (Sereja313) [reply docs/2022-07-18-JSON-RPC-Server-API.md]: I removed part of the module to get a parsing error, looks like I cut off parts of some 2 tokens.

Changed that part to a better example:
https://github.com/runtimeverification/haskell-backend/blob/7add1249814b87deee9adbbaca7a71efcf927701/docs/2022-07-18-JSON-RPC-Server-API.md?plain=1#L444

File: docs/2022-07-18-JSON-RPC-Server-API.md [hunk 3]
```diff
@@ -285,11 +289,14 @@ In some cases, a unifier `condition` for the implication can still be provided,
   "id": 1,
   "method": "simplify",
   "params": {
-    "state": {"format": "KORE", "version": 1, "term": {}}
+    "state": {"format": "KORE", "version": 1, "term": {}},
+    "name": "MODULE-NAME"
```
Comment (ana-pantilie) [inline docs/2022-07-18-JSON-RPC-Server-API.md:293]: From @jberthold : why `name` and not `module`?

**Pull request merged**